mlflow/genai/optimize/optimizers/dspy_mipro_optimizer.py

```diff
@@ -143,8 +143,9 @@ def _get_minibatch_size(
     def _validate_input_fields(self, input_fields: dict[str, type], prompt: Prompt) -> None:
         if missing_fields := set(prompt.variables) - set(input_fields.keys()):
             raise MlflowException(
-                "The following variables of the prompt are missing from "
-                f"the dataset: {missing_fields}"
+                f"Validation failed: Missing prompt variables in dataset "
+                f"Required variables: {missing_fields} "
```

Comment:
Using both `Missing` and `Required` is a bit confusing. Can we adjust the message?